rocksdb/sst_file_iterator.go

```diff
@@ -0,0 +1,196 @@
+package rocksdb
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrChecksumMismatch    = errors.New("Checksum mismatch")
+	ErrMagicNumberMismatch = errors.New("Magic number mismatch")
+	errEnd                 = errors.New("reach end of block")
+)
+
+type SstFileIterator struct {
+	f              *os.File
+	indexBlockIter *blockIterator
+	dataBlockIter  *blockIterator
+	dataBuf        []byte
+	invalid        bool
+	err            error
+	checksumType   ChecksumType
+}
+
+func NewSstFileIterator(f *os.File) *SstFileIterator {
+	return &SstFileIterator{
```

Comment:
I think it's better to load the footer in this function.
`SeekToFirst` may be called multiple times.